src/main/java/frc/robot/utils/CRTGearRatioUtil.java

```diff
@@ -0,0 +1,154 @@
+  }
+
+  /**
+   * 
+   * @param mainGearRotationValue      Rotation value returned by the main gear
+   *                                   encoder
+   * @param secondaryGearRotationValue Rotation value returned by the secondary
+   *                                   gear encoder
+   * @return Returns the number of full rotations of the main gear
+   */
+  public int getMainGearFullRotations(double mainGearRotationValue, double secondaryGearRotationValue) {
+    double k = kFactor();
+    return (int) Math.round(mod(mainGearRotationRatioTerm, //mod the main gear rotation ratio term by the whole thing plus the simplified term to make 0 full rotations work properly
+        mod(k * //this is the multiplier, see the comment in the returns statement of kFactor() for more details
+        (mainGearRotationValue * secondaryGearRotationRatioTermSimplfied
+            - secondaryGearRotationValue * mainGearRotationRatioTermSimplified) //this returns a decimal where the multiplier is different for every gear ratio
+            , mainGearRotationRatioTermSimplified) //mod it with the main gear rotation ratio term so it is within the correct range
+            + mainGearRotationRatioTermSimplified)); //for more information, see the comment in the kFactor() function
+  }
+
+  /**
+   * 
+   * @return Returns true if the gear ratio works with the neccessary number of
+   *         max rotations, false otherwise
+   */
+  public boolean validateGearRatioWithMaxRotations() {
+    boolean works = true;
+    for (int i = 0; i < maxRotations * 6; i++) {
+      double elevatorPos = i * mainGearCircumference / 6; //set elevator positon (expected position that it will check with) through sixths of a rotation until max rotations is reached
+      double epoc = CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1) > 0.9 ? 0.0
+          : CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1) < 0.1 ? 0.0
+              : CRTGearRatioUtil.mod(elevatorPos / mainGearCircumference, 1); //calculate what the main gear encoder value would be at each position - the ternary operators account for annoying float division rounding problems on either end (0 and 1)
+      if (Math.abs(this.getAbsolutePositionOfMainObject(epoc, //the calculated main gear encoder value
+          CRTGearRatioUtil.mod(
+              elevatorPos / mainGearCircumference * secondaryGearRotationRatioTerm /
+                  mainGearRotationRatioTerm, //the calculated secondary gear encoder value
+              1))
```

Comment:
Why does this not need the weird "accounting for nearly 0 or 1" that you did above?

Reply:
the reason it was needed above was because numbers like 0.99 would output numbers that rounded to the wrong number of full rotations. here less precision is needed